dist/Tie-File/lib/Tie/File.pm

```diff
@@ -2128,6 +2129,16 @@ then the array C<@gifts> would appear to contain C<("Gold\n",
 "Frankincense\n", "Myrrh\n")>, or (on Win32 systems) C<("Gold\r\n",
 "Frankincense\r\n", "Myrrh\r\n")>.
 
+=head2 C<binmode>
+
+Specify the binmode to set for files opened by Tie::File.
+
+  # make sure Perl is expecting unicode files and strings
+  use utf8;
+  use feature 'unicode_strings';
+  # tie a file with UTF-8 binmode
+  tie my @array, 'Tie::File', 'array.file', 'binmode' => ':encoding(UTF-8)';
```

Comment:
One of our porting tests, `t/porting/podcheck.t`, imposes a 78-character line length in documentation and calculates tab expansion as part of that process.  This line is slightly over that calculated maximum.  A simple way to fix that would be to break the line in two at an appropriate point, *e.g.,*:
```
diff --git a/dist/Tie-File/lib/Tie/File.pm b/dist/Tie-File/lib/Tie/File.pm
index 9b41efa08a..b45aea9685 100644
--- a/dist/Tie-File/lib/Tie/File.pm
+++ b/dist/Tie-File/lib/Tie/File.pm
@@ -2137,7 +2137,8 @@ Specify the binmode to set for files opened by Tie::File.
   use utf8;
   use feature 'unicode_strings';
   # tie a file with UTF-8 binmode
-  tie my @array, 'Tie::File', 'array.file', 'binmode' => ':encoding(UTF-8)';
+  tie my @array,
+    'Tie::File', 'array.file', 'binmode' => ':encoding(UTF-8)';
 
 =head2 C<mode>
 
```
Here, too, run `make test_porting` before re-pushing.  Thanks.

Reply:
Also, I suggest changing the Subject of this GH pull request to be: "Tie-File: add option to specify binmode".  You can trim the first line of the commit message to simply that as well.

Reply:
Once I delved into the open issue #17494, I determined that this pull request only works for reading files and have work in another branch https://github.com/brainbuz/perl5/tree/trytofix17494, I was able to improve behavior but writes of new data are still 1 character off.
I'd like to wait a few days to see if I can solve this problem, if I can't I'll make the minor changes and add to the doc a warning that with non-byte files Tie::File should be used read only. If I can fix it I'll merge it to this pull request.

Reply:
I see *no* added value in adding "recommended" `use` statements in documentation if there is no relation to the stuff being documented.
`use utf8;` is about the source code and has no relation whatsoever to the file being tied, so it distracts from what is documented. My first though is "why do I need this?"
I'd say leave out what is not required.